### PR TITLE
Feature: height and fullscreen option

### LINKED ISSUE #29: Fullscreen option
What about add `fullscreen` option  for create a fullscreen box?

### PR DESCRIPTION
**Pull request by Caesarovich**
In answer to #29 , here I have added an height constraint option and a fullscreen option.

Example with height option:

```js
import boxen from 'boxen';

console.log(boxen('foo bar', {height: 5}));
// ┌───────┐
// │foo bar│
// │       │
// │       │
// └───────┘
```

Sorry for the wait, I have found a job and couldn't find many time to focus on this issue.

---

Fixes #29

### CODE REVIEW

**Comment by sindresorhus** [comment]
// @LitoMore

**Comment by Caesarovich** [comment]
@sindresorhus Should we really wait for @LitoMore ? The features are there and working as expected ;)

**Comment by LitoMore** [comment]
> @sindresorhus Should we really wait for @LitoMore ? The features are there and working as expected ;)

I will take a look.

**Comment by LitoMore** [comment]
Please add some tests for the `fullscreen` option.

BTW, as we discussed before at https://github.com/sindresorhus/boxen/issues/29#issuecomment-950273311, the `fullscreen` option should be able to accept a function for modifications.

File: index.d.ts [hunk 1]
```diff
@@ -193,6 +193,33 @@ export interface Options {
...
+	// └───────┘
+	```
+	*/
+	readonly height?: number;
+
+	/**
+	If possible, the box will fit all available space within the terminal.
+
+	**Note*: This option overrides the `width` and `height` options.
+
```
Comment (LitoMore) [inline index.d.ts:220]:
```suggestion
```

Comment (LitoMore) [reply index.d.ts:220]: I think `width` and `height` should have higher priority. Then users can create a full-width or full-height box without difficulty.

**Comment by LitoMore** [comment]
Readme needs to be updated.

File: index.d.ts [hunk 2]
```diff
@@ -193,6 +193,33 @@ export interface Options {
 	```
 	*/
 	readonly width?: number;
+
+	/**
+	Set a fixed height for the box.
+
+	**Note*: This option will crop overflowing content.
```
Comment (sindresorhus) [inline index.d.ts:200]:
```suggestion
	__Note__: This option will crop overflowing content.
```

**Comment by Caesarovich** [comment]
@LitoMore I'm wondering if having the possibility to use a function as argument isn't too much complexity for little benefits

**Comment by LitoMore** [comment]
> I'm wondering if having the possibility to use a function as argument isn't too much complexity for little benefits


See https://github.com/sindresorhus/boxen/issues/29#issuecomment-950273311.

**Comment by Caesarovich** [comment]
// @LitoMore

File: index.d.ts [hunk 3]
```diff
@@ -193,6 +193,42 @@ export interface Options {
...
+	// │foo bar│
+	// │       │
+	// │       │
+	// └───────┘
+	```
+	*/
+	readonly height?: number;
+
+	/**
+	__boolean__: Wether or not to fit all available space within the terminal.
```
Comment (sindresorhus) [inline index.d.ts:217]: index.d.ts and readme should be in sync.

File: index.d.ts [hunk 4]
```diff
@@ -193,6 +193,42 @@ export interface Options {
...
+
+	__function__: Pass a callback function to control box dimensions.
```
Comment (sindresorhus) [inline index.d.ts:219]: This should be clearer about it being used to **modify** the calculated dimensions. And also some use-cases for it.

// @LitoMore

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by Caesarovich** [comment]
Should this be considered done ? :) @sindresorhus

**Comment by sindresorhus** [comment]
Looks good :)

**Pull request merged**